lib/Expr/ExprUtil.cpp

```diff
@@ -104,6 +104,21 @@ class SymbolicObjectFinder : public ExprVisitor {
     : objects(_objects) {}
 };
 
+ExprVisitor::Action ConstantArrayFinder::visitRead(const ReadExpr &re) {
+  const UpdateList &ul = re.updates;
+
+  // FIXME should we memo better than what ExprVisitor is doing for us?
+  for (const UpdateNode *un = ul.head; un; un = un->next) {
+    visit(un->index);
+    visit(un->value);
+  }
+
+  if (ul.root->isConstantArray()) {
+    results.insert(ul.root);
+  }
+
+  return Action::doChildren();
```

Comment:
Given that you visit the index you should return `Action::skipChildren()`.

Reply:
Oops. Ignore that. You're visiting the `index` of the update list and not the `index` of the `ReadExpr`, so the code is fine.